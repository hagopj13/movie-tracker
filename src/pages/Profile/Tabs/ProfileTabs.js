// @flow
import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

import TabPanel from 'components/TabPanel/TabPanel';

import FavortiesTab from './Favorites/ProfileFavoritesTab';
import WatchlistTab from './Watchlist/ProfileWatchlistTab';

const useStyles = makeStyles((theme) => ({
  tab: {
    fontSize: theme.typography.body1.fontSize,
  },
}));

const ProfileTabs = () => {
  const [currentTab, setCurrentTab] = useState(0);
  const classes = useStyles();

  const handleTabChange = (event, newTab) => {
    setCurrentTab(newTab);
  };

  return (
    <>
      <Paper square>
        <Tabs
          value={currentTab}
          indicatorColor="primary"
          textColor="primary"
          onChange={handleTabChange}
          centered
        >
          <Tab className={classes.tab} label="Favorites" disableRipple />
          <Tab className={classes.tab} label="Watchlist" disableRipple />
        </Tabs>
      </Paper>
      <TabPanel value={currentTab} index={0}>
        <FavortiesTab />
      </TabPanel>
      <TabPanel value={currentTab} index={1}>
        <WatchlistTab />
      </TabPanel>
    </>
  );
};

export default ProfileTabs;
