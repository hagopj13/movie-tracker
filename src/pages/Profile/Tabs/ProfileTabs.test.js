import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import ProfileTabs from './ProfileTabs';

describe('ProfileTabs component', () => {
  let shallow;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    wrapper = shallow(<ProfileTabs />);
  });

  it('should render the ProfileTabs component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
