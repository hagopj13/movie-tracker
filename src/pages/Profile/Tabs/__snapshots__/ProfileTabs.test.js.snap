// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTabs component should render the ProfileTabs component correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Paper))
    square={true}
  >
    <WithStyles(ForwardRef(Tabs))
      centered={true}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        className="makeStyles-tab-1"
        disableRipple={true}
        label="Favorites"
      />
      <WithStyles(ForwardRef(Tab))
        className="makeStyles-tab-1"
        disableRipple={true}
        label="Watchlist"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Paper))>
  <TabPanel
    index={0}
    value={0}
  >
    <Connect(ProfileFavoritesTab) />
  </TabPanel>
  <TabPanel
    index={1}
    value={0}
  >
    <Connect(ProfileWatchlistTab) />
  </TabPanel>
</Fragment>
`;
